src/wasm-interpreter.h

```diff
@@ -2369,31 +2379,6 @@ class ConstantExpressionRunner : public ExpressionRunner<SubType> {
     NOTE_ENTER("Rethrow");
     return Flow(NONCONSTANT_FLOW);
   }
-  Flow visitStringMeasure(StringMeasure* curr) {
-    return Flow(NONCONSTANT_FLOW);
-  }
-  Flow visitStringEncode(StringEncode* curr) { return Flow(NONCONSTANT_FLOW); }
-  Flow visitStringConcat(StringConcat* curr) { return Flow(NONCONSTANT_FLOW); }
-  Flow visitStringEq(StringEq* curr) { return Flow(NONCONSTANT_FLOW); }
```

Comment:
This is the critical line for `string.eq` - this line overrode the parent, so we always returned "give up"...

Reply:
ohhh got it.